mackerel-plugin-aws-ec2-ebs/lib/aws-ec2-ebs.go

```diff
@@ -42,62 +46,74 @@ var io1Graphs = append([]string{
 type cloudWatchSetting struct {
 	MetricName string
 	Statistics string
-	CalcFunc   func(float64, float64) float64
+	CalcFunc   func(float64) float64
+}
+
+func value(val float64) float64 {
+	return val
+}
+
+func valuePerSec(val float64) float64 {
+	return val / aggregationPeriod
+}
+
+func sec2msec(val float64) float64 {
+	return val * 1000
 }
 
 // http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/monitoring-volume-status.html
 var cloudwatchdefs = map[string](cloudWatchSetting){
 	"ec2.ebs.bandwidth.#.read": cloudWatchSetting{
 		MetricName: "VolumeReadBytes", Statistics: "Sum",
-		CalcFunc: func(val float64, period float64) float64 { return val / period },
+		CalcFunc: valuePerSec,
 	},
 	"ec2.ebs.bandwidth.#.write": cloudWatchSetting{
 		MetricName: "VolumeWriteBytes", Statistics: "Sum",
-		CalcFunc: func(val float64, period float64) float64 { return val / period },
+		CalcFunc: valuePerSec,
 	},
 	"ec2.ebs.throughput.#.read": cloudWatchSetting{
 		MetricName: "VolumeReadOps", Statistics: "Sum",
-		CalcFunc: func(val float64, period float64) float64 { return val / period },
+		CalcFunc: valuePerSec,
 	},
 	"ec2.ebs.throughput.#.write": cloudWatchSetting{
 		MetricName: "VolumeWriteOps", Statistics: "Sum",
-		CalcFunc: func(val float64, period float64) float64 { return val / period },
+		CalcFunc: valuePerSec,
 	},
 	"ec2.ebs.size_per_op.#.read": cloudWatchSetting{
 		MetricName: "VolumeReadBytes", Statistics: "Average",
-		CalcFunc: func(val float64, period float64) float64 { return val },
+		CalcFunc: value,
 	},
 	"ec2.ebs.size_per_op.#.write": cloudWatchSetting{
 		MetricName: "VolumeWriteBytes", Statistics: "Average",
-		CalcFunc: func(val float64, period float64) float64 { return val },
+		CalcFunc: value,
 	},
 	"ec2.ebs.latency.#.read": cloudWatchSetting{
 		MetricName: "VolumeTotalReadTime", Statistics: "Average",
-		CalcFunc: func(val float64, period float64) float64 { return val * 1000 },
+		CalcFunc: sec2msec,
 	},
 	"ec2.ebs.latency.#.write": cloudWatchSetting{
 		MetricName: "VolumeTotalWriteTime", Statistics: "Average",
-		CalcFunc: func(val float64, period float64) float64 { return val * 1000 },
+		CalcFunc: sec2msec,
 	},
```

Comment:
**VolumeTotalReadTime** and **VolumeTotalWriteTime** is in seconds, but this plugin's graph defs is labeled `(ms/op)`.
https://github.com/mackerelio/mackerel-agent-plugins/blob/1be28e2aed5f8ca05cf8b2ee35717283405c722b/mackerel-plugin-aws-ec2-ebs/lib/aws-ec2-ebs.go#L129-L136